plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -313,8 +373,54 @@ open class DefaultPageCreator(
                     }
                 }
             }
-            group(styles = setOf(ContentStyle.TabbedContent), sourceSets = mainSourcesetData + extensions.sourceSets) {
-                val csWithConstructor = classlikes.filterIsInstance<WithConstructors>()
+
+            val csWithConstructor = classlikes.filterIsInstance<WithConstructors>()
+            val scopes = documentables.filterIsInstance<WithScope>()
+
+            val containsAnyConstructor = csWithConstructor.isNotEmpty() && documentables.shouldRenderConstructors()
+            val containsAnyMember = containsAnyConstructor || scopes.any { it.classlikes.isNotEmpty() || it.functions.isNotEmpty() || it.properties.isNotEmpty() }
+            val extraTabs = ExtraTabs(
+                listOfNotNull(
+                    if (!containsAnyMember) null else ContentTab(
+                        ContentText(
+                            "Members",
+                            DCI(mainDRI, ContentKind.Main),
+                            sourceSets = (mainSourcesetData + extensions.sourceSets).toDisplaySourceSets()
+                        ),
+                        listOf(
+                            "Constructors",
+                            "Types",
+                            "Functions",
+                            "Inherited Functions",
+                            "Properties",
+                            "Inherited Properties"
+                        )
+                    ),
+                    if (extensions.isEmpty()) null else ContentTab(
+                        ContentText(
+                            "Members + Extensions",
+                            DCI(mainDRI, ContentKind.Main),
+                            sourceSets = (mainSourcesetData + extensions.sourceSets).toDisplaySourceSets()
+                        ),
+                        listOf(
+                            "Constructors",
+                            "Types",
+                            "Functions",
+                            "Inherited Functions",
+                            "Properties",
+                            "Inherited Properties",
+                            "Extensions"
+                        )
+                    )
+
+                )
+            )
+
```

Comment:
Once you understand how it works, it makes total sense. However, I couldn't understand in the beginning how it would add match content and the tab. I think it would've helped if tab content was typed and I could just look at usages.

With what I proposed above, it would look something like

```kotlin
val membersTab = ContentTab(
    name = "Members",
    contentTypes = listOf(
        TabbedContentType.TYPE,
        TabbedContentType.CONSTRUCTOR,
        TabbedContentType.FUNCTION,
        TabbedContentType.PROPERTY,
        TabbedContentType.ENTRY,
    )
)

val membersWithExtensionsTab = ContentTab(
    name = "Members & Extensions",
    contentTypes = membersTab.contentTypes + TabbedContentType.EXTENSION
)
```